File: contacting-conf-web/src/main/scala/nl/ing/api/contacting/conf/domain/BlacklistItemVO.scala
package nl.ing.api.contacting.conf.domain


import com.ing.api.contacting.dto.resource.blacklist.BlacklistFunctionality.BlacklistFunctionality
import com.ing.api.contacting.dto.resource.blacklist.BlacklistType.BlacklistType

import java.time.LocalDateTime

case class BlacklistItemVO(id: Option[Long],
                           functionality: BlacklistFunctionality,
                           entityType: BlacklistType,
                           value: String,
                           startDate: LocalDateTime,
                           endDate: Option[LocalDateTime])

File: contacting-conf-web/src/main/java/nl/ing/api/contacting/conf/domain/model/blacklist/BlacklistItemVO.java
package nl.ing.api.contacting.conf.domain.model.blacklist;

import com.ing.api.contacting.dto.java.resource.blacklist.BlacklistFunctionality;
import com.ing.api.contacting.dto.java.resource.blacklist.BlacklistType;

import java.time.LocalDateTime;
import java.util.Optional;

public record BlacklistItemVO(
        Long id,
        BlacklistFunctionality functionality,
        BlacklistType entityType,
        String value,
        LocalDateTime startDate,
        Optional<LocalDateTime> endDate
) {

    public Optional<Long> getId() {
        return Optional.ofNullable(id);
    }

    public Optional<LocalDateTime> getEndDate() {
        return endDate;
    }
}

